src/components/DisplayArticlesData.jsx

```diff
@@ -1,60 +1,43 @@
-import React, { useState, useEffect } from 'react';
-import { getArticleData } from '../modules/articlesData';
+import React, { useEffect } from 'react';
+import { getArticleData, getSpecificArticle } from '../modules/articlesData';
+import ArticleCard from './ArticleCard';
+import SingleArticle from './SingleArticle';
+import { useSelector, useDispatch } from 'react-redux';
 import { Card } from 'semantic-ui-react';
 
 const DisplayArticlesData = () => {
-  const [articlesData, setArticlesData] = useState([]);
+  const dispatch = useDispatch();
+  const articles = useSelector((state) => state.articlesIndex);
+  const singleArticle = useSelector((state) => state.singleArticle);
+
   const getArticles = async () => {
-    let result = await getArticleData();
-    setArticlesData(result.data.articles);
+    const result = await getArticleData();
+    dispatch({ type: 'SET_ARTICLES_INDEX', payload: result.data.articles });
+    dispatch({ type: 'SET_SINGLE_ARTICLE' });
+  };
+
+  const fetchSingleArticle = async (event) => {
+    const id = event.target.dataset.id;
+    const response = await getSpecificArticle(id);
+    dispatch({ type: 'SET_SINGLE_ARTICLE', payload: response.data.article });
   };
 
   useEffect(() => {
     getArticles();
   }, []);
 
-  let articleIndex = articlesData.map((article) => {
-    return (
-      <Card key={article.id} data-cy={`article-${article.id}`}>
-        <Card.Content>
-          <Card.Header
-            data-cy="title"
-            style={{
-              fontWeight: 'bold',
-              fontSize: 30,
-              color: 'black',
-              textAlign: 'center',
-            }}
-          >
-            {article.title}
-          </Card.Header>
-          <Card.Meta
-            data-cy="sub-title"
-            style={{
-              fontWeight: 'bold',
-              fontSize: 20,
-              color: 'black',
-              textAlign: 'center',
-            }}
-          >
-            {article.sub_title}
-          </Card.Meta>
-          <Card.Description
-            data-cy="content"
-            style={{ fontWeight: 'small', fontSize: 15, color: 'black' }}
-          >
-            {article.content}
-          </Card.Description>
-          <Card.Meta data-cy="author-email">{article.author}</Card.Meta>
-          <Card.Meta data-cy="created-at">{`Created at: ${article.created_at}`}</Card.Meta>
-          <Card.Meta data-cy="updated-at">{`Updated at: ${article.updated_at}`}</Card.Meta>
-        </Card.Content>
-      </Card>
-    );
-  });
-
   return (
-    <>{articlesData.length > 0 && <div data-cy="index">{articleIndex}</div>}</>
+    <>
+      {articles.length > 0 && !singleArticle && (
+        <Card.Group data-cy="index" itemsPerRow={3}>
+          <ArticleCard
+            articles={articles}
+            fetchSingleArticle={fetchSingleArticle}
+          />
+        </Card.Group>
+      )}
+      <SingleArticle singleArticle={singleArticle} getArticles={getArticles} />
```

Comment:
You can subscribe to the state of a single article in the component instead of passing it as props

Reply:
We also refactored this one too.